packages/kilt-did/src/DidResolver/DefaultResolver.ts

```diff
@@ -44,44 +63,142 @@ async function detailsFromIdentifier(
       capabilityDelegationKey,
     ]
   }
-  const didDetails: DidDetailsCreationOpts = {
-    did: getKiltDidFromIdentifier(identifier),
-    keys: publicKeys,
-    keyRelationships,
-    lastTxIndex: lastTxCounter.toBn(),
-  }
+
+  let services: IServiceDetails[] = []
   if (servicesResolver && endpointData) {
     const { contentHash, contentType, urls } = endpointData
-    didDetails.services = await servicesResolver(contentHash, urls, contentType)
+    services = await servicesResolver(contentHash, urls, contentType)
   }
-  return new DidDetails(didDetails)
+
+  return new FullDidDetails({
+    did: getKiltDidFromIdentifier(identifier, 'full', version),
+    keys: publicKeys,
+    keyRelationships,
+    lastTxIndex: lastTxCounter.toBn(),
+    services,
+  })
 }
 
-export async function resolveKey(
-  didUri: string
-): Promise<IDidKeyDetails | null> {
-  const { did, fragment } = parseDidUrl(didUri)
-  return queryKey(did, fragment)
+/**
+ * Resolve a DID URI (including a key ID or a service endpoint ID).
+ *
+ * @param didUri The DID URI to resolve.
+ * @param opts The options to provide to the service resolver.
+ * @returns The DID, key, or service endpoint details depending on the input URI. If not resource can be resolved, null is returned.
+ */
+export async function resolve(
+  didUri: string,
+  opts: ResolverOpts = {}
+): Promise<IDidDetails | IDidKeyDetails | IServiceDetails | null> {
+  const { identifier, type, version, fragment, encodedDetails } = parseDidUrl(
+    didUri
+  )
+
+  switch (type) {
+    case 'full': {
+      const details = await queryFullDetailsFromIdentifier(
+        identifier,
+        opts,
+        version
+      )
+      // If the URI is a subject DID, return the retrieved details.
+      if (!fragment) {
+        return details
+      }
+
+      // Otherwise, return either the key or the service endpoints referenced by the URI.
+      return details?.getKey(didUri) || details?.getService(didUri) || null
+    }
+    case 'light': {
+      // In light DIDs the key type encoding (first two chars) is part of the identifier.
+      // We are sure the URI follows the expected structure as it has been checked in `parseDidUrl`.
+      const keyTypeEncoding = identifier.substring(0, 2)
```

Comment:
S: Since we have a function which builds a `FullDidDetails` object above, I think it would be good to have something similar here. Like a `buildLightDetailsFromIdentifier` function e.g.

Reply:
Fixed in https://github.com/KILTprotocol/sdk-js/pull/408/commits/ed4d9c3448b4bfc329e87bdd5930b88f88287e64.